File: kimchanho/16week/BOJ_12605.js
const solution = (input) => {
  const N = Number(input[0]);
  for (let i = 1; i <= N; i++) {
    const words = input[i].split(" ");
    let result = "";
    while (words.length) {
      result += words.pop() + " ";
    }
    console.log(`Case #${i}: ${result}`);
  }
};

// setting for baekjoon online judge
const inputArr = [];
require("readline")
  .createInterface({ input: process.stdin, output: process.stdout })
  .on("line", (line) => inputArr.push(line))
  .on("close", () => solution(inputArr));

File: sehun/16week/BOJ_12605.py
def pop_and_push(input_string: str) -> str:
    # 파이썬에서 스택 모듈?은 없지만 리스트의 append(push)와 pop메소드를 이용가능... 성능은 조사해봐야함
    input_array = input_string.split()
    stack = []
    result = ""
    for word in input_array:
        stack.append(word)

    for i in range(len(stack)):
        if i == 0:
            pass
        else:
            result += " "
        result += stack.pop()
    return result


n = int(input())

for i in range(1, n + 1):
    print(f"Case #{i}: {pop_and_push(input())}")
